diag/helpers.go

```diff
@@ -0,0 +1,36 @@
+package diag
+
+import "fmt"
+
+// FromErr will convert an error into a Diagnostics. This returns Diagnostics
+// as the most common use case in Go will be handling a single error
+// returned from a function.
+//
+//   if err != nil {
+//     return diag.FromErr(err)
+//   }
+func FromErr(err error) Diagnostics {
```

Comment:
Overall I do like this pattern, however I would still like a convenience to create a singular `Diagnostic` from an error. would it be crazy to keep the helpers for creating on a single diag, but then have a helper to quickly convert to the collection?
```
return diag.FromErr(err).ToDiags()
```

Reply:
It's valuable in scenarios where we have gathered warnings, and encounter a function that just returns a go error
```
if err != nil {
    return append(diags, diag.FromErr(err)
}
```

Reply:
I think the thing is, that you are burdening the 80% case for the 20% case, and you could easily rewrite that second one as:

```go
if err != nil {
    return append(diags, diag.FromErr(err)...)
}
```

Reply:
I wish the package was called `diags` with this model, but that would collide with calling your slice `diags`.